src/Controller/Webhook/LabelCreated.php

```diff
@@ -0,0 +1,140 @@
+
+use Boekuwzending\Magento\Service\OrderShipperInterface;
+use Boekuwzending\Magento\Utils\Constants;
+use Exception;
+use JsonException;
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Controller\Result\Json;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NotFoundException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * This /labelCreated webhook gets called by the platform when a label is made for one of our orders.
+ *
+ * @package Boekuwzending\Magento\Controller\Webhook
+ */
+class LabelCreated extends WebhookBase implements HttpPostActionInterface
+{
+    /**
+     * @var OrderShipperInterface
+     */
+    private $orderShipper;
+
+    /**
+     * Label constructor.
+     *
+     * @param LoggerInterface $logger
+     * @param RequestInterface $request
+     * @param ScopeConfigInterface $scopeConfig
+     * @param OrderShipperInterface $orderShipper
+     * @param JsonFactory $resultJsonFactory
+     */
+    public function __construct(LoggerInterface $logger,
+                                RequestInterface $request,
+                                JsonFactory $resultJsonFactory,
+                                ScopeConfigInterface $scopeConfig,
+                                OrderShipperInterface $orderShipper)
+    {
+        parent::__construct($logger, $request, $resultJsonFactory, $scopeConfig);
+
+        $this->orderShipper = $orderShipper;
+    }
+
+    /**
+     * Handles the POST /boekuwzending/webhook/labelCreated call.
+     *
+     * @return Json
+     *
+     * @throws LocalizedException
+     * @throws JsonException
+     */
+    public function execute(): Json
+    {
+        $requestBody = $this->deserializeRequestBody();
+
+        if (null === $requestBody) {
+            return $this->badRequest(__('Could not parse JSON request body'));
+        }
+
+        $data = $requestBody['data'];
+
+        // TODO: how to document what body we expect?
+        if (!array_key_exists('entity_id', $data)) {
+            return $this->missingRequestBodyKey('data.entity_id');
+        }
+        if (!array_key_exists('shipment_id', $data)) {
+            return $this->missingRequestBodyKey('data.shipment_id');
+        }
+        if (!array_key_exists('external_order_id', $data)) {
+            return $this->missingRequestBodyKey('data.external_order_id');
+        }
+        if (!array_key_exists('carrier_code', $data)) {
+            return $this->missingRequestBodyKey('data.carrier_code');
+        }
+        if (!array_key_exists('carrier_title', $data)) {
+            return $this->missingRequestBodyKey('data.carrier_title');
+        }
+        if (!array_key_exists('tracking_number', $data)) {
+            return $this->missingRequestBodyKey('data.tracking_number');
+        }
```

Comment:
```suggestion
        if (!array_key_exists('entity_id', $data)) {
            return $this->missingRequestBodyKey('data.entity_id');
        }
        if (!array_key_exists('shipment_id', $data)) {
            return $this->missingRequestBodyKey('data.shipment_id');
        }
        if (!array_key_exists('external_order_id', $data)) {
            return $this->missingRequestBodyKey('data.external_order_id');
        }
        if (!array_key_exists('carrier_code', $data)) {
            return $this->missingRequestBodyKey('data.carrier_code');
        }
        if (!array_key_exists('carrier_title', $data)) {
            return $this->missingRequestBodyKey('data.carrier_title');
        }
        if (!array_key_exists('tracking_number', $data)) {
            return $this->missingRequestBodyKey('data.tracking_number');
        }
```

!arra... vervangen in false === arra...